src/controller_examples/fluent_controller/fluentbit/exec/resource/service.rs

```diff
@@ -156,4 +164,55 @@ pub fn make_service(fb: &FluentBit) -> (service: Service)
     service
 }
 
+fn make_new_ports(ports: Vec<ContainerPort>) -> (service_ports: Vec<ServicePort>)
+    ensures service_ports@.map_values(|p: ServicePort| p@) == ports@.map_values(|p: ContainerPort| model_resource::make_service_port(p@)),
+{
+    let mut service_ports = Vec::new();
+    let mut i = 0;
+    proof {
+        assert_seqs_equal!(
+            service_ports@.map_values(|p: ServicePort| p@),
+            Seq::new(i as nat, |k: int| model_resource::make_service_port(ports[k]@))
+        );
+    }
+    while i < ports.len()
+        invariant
+            i <= ports@.len(),
+            service_ports@.map_values(|p: ServicePort| p@) == Seq::new(i as nat, |k: int| model_resource::make_service_port(ports[k]@)),
+        ensures
+            service_ports@.map_values(|p: ServicePort| p@) == Seq::new(ports@.len(), |k: int| model_resource::make_service_port(ports[k]@)),
```

Comment:
I am a bit surprised that you don't need to write `decrease` here. Did I miss anything?

Reply:
Seems that the tutorial does not even have that https://verus-lang.github.io/verus/guide/while.html

Reply:
Not sure I follow. But I thought `decreases` only exists in spec functions. And there's no recursion.

Reply:
I am referring to this https://dafny.org/latest/DafnyRef/DafnyRef#sec-loop-termination
But I guess Verus does not support that yet

Reply:
In this case the `i` is decreasing? But in our case `i` is increasing.

Reply:
`ports@.len() - i`

Reply:
You can try to add `decrease` clause to this loop and see what Verus will say